core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -49,7 +50,7 @@ import scala.util.control.{ControlThrowable, NonFatal}
  *   Acceptor has N Processor threads that each have their own selector and read requests from sockets
  *   M Handler threads that handle requests and produce responses back to the processor threads for writing.
  */
-class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time) extends Logging with KafkaMetricsGroup {
+class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time, val credentialCache: Option[ScramCredentialCache] = None) extends Logging with KafkaMetricsGroup {
```

Comment:
I have made the cache more generic since delegation tokens will also require a cache, but with credentials of a different type. Have moved authentication related code into a separate class.